From 31e331789bb64b78ec0496b53ee80c615bfc7563 Mon Sep 17 00:00:00 2001
From: cnfatal <cnfatal@gmail.com>
Date: Mon, 16 Feb 2026 01:24:05 +0800
Subject: [PATCH 1/3] Add 16K page alignment support for Google Play

Add -Wl,-z,max-page-size=16384 linker flag to ensure all .so files
are 16K page aligned, as required by Google Play.

- Application.mk: add APP_LDFLAGS for ndk-build linked libraries
- common.sh: add to LDFLAGS for Python/Cython module linking
---
 native/jni/Application.mk | 1 +
 native/scripts/common.sh  | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/native/jni/Application.mk b/native/jni/Application.mk
index 4330b0c..4a85b67 100644
--- a/native/jni/Application.mk
+++ b/native/jni/Application.mk
@@ -5,5 +5,6 @@ APP_MODULES := python SDL2 png16 jpeg libwebp SDL2_image SDL2_gfx SDL2_ttf SDL2_
 APP_ABI := $(ARCH)
 APP_STL := c++_shared
 APP_CFLAGS += $(OFLAG)
+APP_LDFLAGS += -Wl,-z,max-page-size=16384
 
 APP_PLATFORM := $(ANDROID_PLATFORM)
\ No newline at end of file
diff --git a/native/scripts/common.sh b/native/scripts/common.sh
index 71e950d..dfb9d3c 100644
--- a/native/scripts/common.sh
+++ b/native/scripts/common.sh
@@ -60,7 +60,7 @@ activate_toolchain () {
     export RANLIB="$GCC_ARCH-ranlib"
 
     export CFLAGS="-DANDROID -D__ANDROID_API__=${ANDROID_PLATFORM#android-} $PICFLAG"
-    export LDFLAGS="$PICFLAG"
+    export LDFLAGS="$PICFLAG -Wl,-z,max-page-size=16384"
 
     libdir "$INSTALLDIR/lib"
     libdir "$NATIVE/obj/local/$PLATFORM"
-- 
2.34.1

